client/state/data-layer/wpcom/sites/admin-menu/index.js

```diff
@@ -17,8 +18,38 @@ export const requestFetchAdminMenu = ( action ) =>
 		action
 	);
 
-export const handleSuccess = ( { siteId }, menuData ) => {
-	return receiveAdminMenu( siteId, menuData );
+const sanitizeUrl = ( url, site ) => {
+	if ( new RegExp( `^https?://${ site?.domain }` ).test( url ) ) {
+		return url;
+	} else if ( /^\//.test( url ) ) {
+		return url;
+	}
+
+	return '';
+};
+
+const sanitizeMenuItem = ( menuItem, site ) => {
+	let sanitizedChildren;
+	if ( menuItem.children ) {
```

Comment:
Is it worth checking also that `children` is an Array by using `Array.isArray(menuItem.children)` too?

Reply:
Yup, better safe than sorry in this case! ad1c6a2